Update 0004000000174600.txt

TitunAVC opened the pull request:
<!--- Provide a general summary of the issue in the Title above, prefixed with (CheatFolder Name) -->

## Description
Added multiple cheats that I have been able to recover from pages and another that I have had to modify from them.
I have tried most and they work on citra.


## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Cheat fix
- [x] New cheats
- [ ] Regional fix / Addition
- [ ] Other

## Additional Notes
In some, different ids are needed, which are found in GBAtemp. I don't know if we could add them here or put a link that goes to their thread.

TitunAVC committed:
Update 0004000000174600.txt

TitunAVC committed:
Update 0004000000174600.txt

iSharingan commented:
the code "Item quantities x99" already does the same thing as the final code you added and applies to all items (including in shops/bag/etc) without the janky activation conditions. the code you added only affects the first item slot - and by changing the item ID of that slot it causes the code to be unreliable on updating the correct item slot (since it requires the game to auto sort items on an inventory update, which also means you're probably deleting/replacing the item in the first storage slot with a different item using this code).

A better option for a single item edit code would be to add a conditional block to the 99 item code to compare the value at the item ID offset to the desired item ID and only update the quantity if it matches while going through the loop of all storage/bag/shop slots - thus avoiding the problems with the code you supplied.

Pokemon, item, and move IDs can be added in text files to be used as a guide (add a "Guide" folder next to the cheat text file and put the ID list text files in there. when the Guide folder is placed in the correct title ID folder on the 3DS, the plugin will load them to be read in the plugin under the guide tab (iirc the path would be sd:/luma/plugins/ActionReplay/[game title ID]/Guide/[text files go here and can be sorted into categories via folders])

TitunAVC commented:
To be honest, I don't know anything about editing or creating codes, I was only able to edit a small code, that of the movements, because it followed a very simple pattern.
(I collected them from old pages with an old page display program using saves)
The item code is messy, though it's fine to check out a specific item if you don't have it, but you have more to lose than to get it right.
Regarding the folders I put it now.

TitunAVC committed:
Add files via upload


Folder with the IDs of the Pokemon and Movements

TitunAVC committed:
Create Read

Agradecimientos al autor de las IDs

TitunAVC committed:
Delete Read

I didn't put it in TXT

TitunAVC committed:
Add files via upload

Now it is in TXT
Sorry for so many commits I'm still not very good at using github

iSharingan commented:
[ActionReplayCodeTypes.txt](https://github.com/iSharingan/CTRPF-AR-CHEAT-CODES/blob/663f0739c80bac99eabd582adf499aa4ef1a03e4/ActionReplayCodeTypes.txt) is there to help with making/fine tuning codes. A lot of old codes use the wrong value size and can damage data or your save/slowly build up errors until the game crashes. Referring to that guilde when making/trying out a code always helps.

Essentially the 99 item code could be rewritten as (after updating for actual variable sizes [so you don't read/write values you shouldn't. The existing item code works, but only gets away with its bad practices because there happens to be empty bytes int he item format], and adjusting the start address back to point at item ID instead of item quantity)

```
[Set quantity of specific item to 99]
D3000000 0886AD88 (set offset)
C0000000 000001F0 (loop 0x1f0 times)
90000000 0000XXXX (if XXXX equals the value at offset+0 [in this case checking item ID])
10000002 00000063 (set value at offset+2 to 0x0063 [in this case, item quantity -> 99])
D0000000 00000000 (endif)
DC000000 00000040 (offset +0x40 [in this case, move to next item slot])
D1000000 00000000 (loop execute)
{Edit the end of line 3 to be the desired item ID\n\nRequires you to have the intended item in your bag/storage - but also affects the item if it exists in a shop}
```

Edit: look, I'm not trying to block this, I'm offering info to help with your own experience (IE: help reduce the number of bad codes you end up trying/using as well as provide tools to make/edit codes as you like). I'm reopening and applying it. I just wanted to be sure you saw my comment first.

TitunAVC commented:
Yes, I know, and I apologize, I am a very shy person, and it is very difficult for me to talk to people, if I don't understand a subject I get very nervous, and I don't know what to do. For which I apologize again for having closed it.
I have one last question for the creation of the codes, is it necessary to do it from 3ds or is there some method from PC?
And I'm sorry to be so repetitive, but I'm really sorry.
（┬＿┬）

iSharingan commented:
> is it necessary to do it from 3ds or is there some method from PC

I do it on 3DS because the blank CTRPFAR plugin makes it super easy to search for addresses, write codes, and test them as I go - but if you can access and search the game's RAM on Citra, you can do it there as well. You'll likely just need to refer to the code types far more often (the plugin's AR code editor both adds the codes to the list in realtime but also gives a pseudocode translation off to the side to let you see what the code is doing - plus tools to let you convert hex to decimal and vice versa/jump to memory offsets in the memory viewer directly/etc.). Iirc there's a way to get the CTRPFAR plugin running on Citra too, but I lost the exact method as I don't use it. If you want, check out the "Nanquitas' Playground" on Discord (Nanquitas is the one that wrote the CTRPFAR plugin), as the info as well as the latest plugin and code writing tutorials/tips should all be available there. Human interaction is optional, but available if you have questions: https://discord.gg/z4ZMh27

If you need help and want to talk to someone 'more familiar', my Discord username is iShar#1771 (I'm refusing to update to the new Discord naming system, lol) - just let me know where you added me from to help keep track of who added me from online games and who I'm helping with other things.
